src/template/wasm2c.declarations.c

```diff
@@ -120,6 +120,187 @@ DEFINE_STORE(i64_store8, u8, u64)
 DEFINE_STORE(i64_store16, u16, u64)
 DEFINE_STORE(i64_store32, u32, u64)
 
+// SIMD
+
+#if defined(WASM_RT_ENABLE_SIMD)
+
+#ifdef __x86_64__
+#define SIMD_FORCE_READ(var) wasm_asm("" ::"x"(var));
+#else
+#define SIMD_FORCE_READ(var)
+#endif
+// TODO: equivalent constraint for ARM and other architectures
+
+#define DEFINE_SIMD_LOAD_FUNC(name, func, t)              \
+  static inline t name(wasm_rt_memory_t* mem, u64 addr) { \
+    MEMCHECK(mem, addr, t);                               \
+    t result = func((v128*)&mem->data[addr]);             \
+    SIMD_FORCE_READ(result);                              \
+    return result;                                        \
+  }
+
+#define DEFINE_SIMD_LOAD_LANE(name, func, t, lane)               \
+  static inline t name(wasm_rt_memory_t* mem, u64 addr, t vec) { \
+    MEMCHECK(mem, addr, t);                                      \
+    t result = func((v128*)&mem->data[addr], vec, lane);         \
+    SIMD_FORCE_READ(result);                                     \
+    return result;                                               \
+  }
+
+#define DEFINE_SIMD_STORE(name, t)                                    \
+  static inline void name(wasm_rt_memory_t* mem, u64 addr, t value) { \
+    MEMCHECK(mem, addr, t);                                           \
+    simde_wasm_v128_store((v128*)&mem->data[addr], value);            \
+  }
+
+#define DEFINE_SIMD_STORE_LANE(name, func, t, lane)                   \
+  static inline void name(wasm_rt_memory_t* mem, u64 addr, t value) { \
+    MEMCHECK(mem, addr, t);                                           \
+    func((v128*)&mem->data[addr], value, lane);                       \
+  }
+
+DEFINE_SIMD_LOAD_FUNC(v128_load, simde_wasm_v128_load, v128);
+
+DEFINE_SIMD_LOAD_FUNC(v128_load8_splat, simde_wasm_v128_load8_splat, v128);
```

Comment:
Should the "t" param here possibly be a u8 instead of a v128? I don't think it needs to MEMCHECK room for an entire v128 if it's only loading a u8 and then splatting it. And similarly for the load16_splat, load32_splat, load32_zero/load64_zero, loadN_lane, storeN_lane, etc.

Reply:
Yeah, good catch. I just pushed a fix for that.